### DISCUSSION TITLE
No way to limit # of redis connections

### DISCUSSION THREAD

**Discussion by aidan-pf** [celery/celery#8575] [General]
Am I missing something? There appears to be no way to limit the number of redis connections each worker makes. Each worker is making 10 redis connections, no matter what I update in the config.

The only commands that seem to reduce connections (each taking away 1 connection) are `--without-gossip` `--without-mingle` `--without-heartbeat`.

example code:

```
from celery import Celery

my_app = Celery(
    "tasks",
    broker="redis://:key@localhost:6379/0",
    task_acks_late=True,
    task_reject_on_worker_lost=True,
    worker_prefetch_multiplier=1,
    broker_pool_limit=2,
    broker_transport_options={
        "visibility_timeout": 240,
        "fanout_prefix": True,
        "fanout_patterns": True,
        "max_connections": 2,
        "socket_keepalive": True,
    },
    result_expires=60,
    broker_connection_retry_on_startup=True,
    worker_send_task_events=True,
    task_send_sent_event=True,
)


@my_app.task(
    bind=True,
    autoretry_for=(Exception,),
    retry_kwargs={"max_retries": 3, "countdown": 3},
    default_retry_delay=5,
    soft_time_limit=10,
    timeout=240,
)
def process_data(self, request_data):
    pass


# redis-server --requirepass key
# redis-cli -a key info clients

# celery -A src.main.my_app worker --loglevel=debug --concurrency=1 -Q celery
```

This opens 10 connections to redis.

I quickly hit the upper limit of Redis max connections (10,000), when running workers on Kubernetes. Increasing max connections in Redis doesn't help, because the CPU gets overwhelmed with the # of connections.

**Comment by bolorundurovj** [comment]
@aidan-pf did you find a solution?

**Comment by bolorundurovj** [comment]
@aidan-pf did you find a solution?

**Reply by aidan-pf** [reply]
Run fewer celery clients with higher worker concurrency for each client. This will reduce the # of redis connections in sum.
